### DISCUSSION TITLE
both send and receive

### DISCUSSION THREAD

**Discussion by slechtvalk200** [someweisguy/esp_dmx#147] [Q&A] [answered]
can someone handle both send and receive with a m5stack core and DMX module? I can do both separate, but not both in one sketch.

**Comment by bensuffolk** [comment, marked as answer]
I use both send an receive at the same time with a dual core ESP32 using the esp-idf framework so there is no reason why you should be able to. I don't use the Arduino framework so can't comment if thats what you are using.

**Reply by doncogit** [reply]
hi,

right now iam trying to do a heavy task while receiving DMX.
As soon as i implement the receiving code its slowing down my loop.
Are you using both cores to achieve sending and receiving at the same time?
One task per core?

**Reply by bensuffolk** [reply]
All my dmx is running on the same core. I have the tasks as a higher priority than other tasks though.

**Reply by bensuffolk** [reply]
Also worth making sure you don’t block the tasks with processing, put the data in a queue and process it in another task

**Reply by doncogit** [reply]
thanks, i did it without multitasking before, that was the problem. The Read function blocked my code. Now with two tasks it works on one core.

**Reply by someweisguy** [reply]
The DMX read function is a blocking function; it will block for the number of ticks that is passed to it, typically `DMX_TIMEOUT_TICK`. If you'd like to be a non-blocking function, you can pass it a `0` instead. :)

**Reply by doncogit** [reply]
Thanks mitch, i tried that too and it worked but iam not sure which of the methods is the best implementation in terms of performance and reliability.
